hapi-utils/src/main/java/com/hedera/services/throttles/GasLimitBucketThrottle.java

```diff
@@ -0,0 +1,73 @@
+package com.hedera.services.throttles;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+/**
+ * Responsible for throttling transaction by gas limit.
+ * Uses a {@link DiscreteLeakyBucket} under the hood.
```

Comment:
We do use buckets under the hood in both BucketThrottle (opsPerSec) and GasLimitBucketThrottle, but in very different ways and merging them might be strange and ambiguous, because there's a conceptual difference between them 
- Ops per sec BucketThrottles rate transactions based on functionality type and assign a weight coefficient to them based on how many transactions of one type can fit in the bucket compared to transactions of another type, with each transaction type requiring a known amount of free space in the bucket from then on and using a formula to calculate space required
- With GasLimitBucketThrottles the 'weighting' of the transaction and calculations of space required in the bucket happens at runtime and it's not possible to know the weight of the transaction in advance, also the formula is much simpler with the current implementation, because it simply compares the TX gasLimit and the bucket free capacity 

That said, I think we can remove the fields from the protobufs and configure them in global dynamic props, this way we'll be open to changes in the future.